### PR TITLE
First cut at logs bicep doc updates

### PR DESCRIPTION
**Pull request by lluppesms**
Updated the Azure.KeyVault.Logs.md with Bicep that mimics the ARM code in it.

Tried to do the same with Azure.FrontDoor.Logs.md, but ran into lots of issues. There was no ARM example code, so I created an template and deployed a simple Front Door. Tried to add the logging for FrontdoorAccessLog and FrontdoorWebApplicationFirewallLog settings to the ARM and Bicep, but I can't find the syntax to automate that setting. All the docs I've found just explain how to set it manually. Spent way too much time today searching for this answer...!

Is this one of those where there just isn't an ARM setting yet, so the docs should just say to do it manually...?  Any idea? or who would know where else to look...?

### CODE REVIEW

File: docs/en/rules/Azure.FrontDoor.Logs.md [hunk 1]
```diff
@@ -25,6 +25,103 @@ Management operations for Front Door is captured automatically within Azure Acti
...
+
+resource frontDoorResource 'Microsoft.Cdn/profiles@2021-06-01' = {
+  name: frontDoorName
+  location: 'Global'
+  sku: {
+    name: 'Standard_AzureFrontDoor'
+  }
+}
+
+resource frontDoorInsightsResource 'Microsoft.Cdn/profiles/providers/diagnosticSettings@2020-01-01-preview' = {
```
Comment (BernieWhite) [inline docs/en/rules/Azure.FrontDoor.Logs.md:80]: Try these updates.

```suggestion
resource frontDoorInsightsResource 'Microsoft.Insights/diagnosticSettings@2020-01-01-preview' = {
```

File: docs/en/rules/Azure.FrontDoor.Logs.md [hunk 2]
```diff
@@ -25,6 +25,103 @@ Management operations for Front Door is captured automatically within Azure Acti
...
+  name: '${frontDoorName}/Microsoft.Insights/service'
+  dependsOn: [
+    frontDoorResource
+  ]
```
Comment (BernieWhite) [inline docs/en/rules/Azure.FrontDoor.Logs.md:84]: Try these updates.

```suggestion
  scope: frontDoorResource
  name: 'service'
```

File: docs/en/rules/Azure.FrontDoor.Logs.md [hunk 3]
```diff
@@ -25,6 +25,103 @@ Management operations for Front Door is captured automatically within Azure Acti
...
+Deploy Test Script: For debugging purposes, save the bicep above to sampleFrontDoor.bicep and run this command to test the deploy of this template:
+
+```ps1
+az deployment group create -n front-door-test-20220919T150200Z --resource-group rg_sandbox --template-file 'sampleFrontDoor.bicep' --parameters frontDoorName=yourFrontDoorName workSpaceId=/subscriptions/yourSubscriptionId/resourcegroups/yourResourceGroup/providers/microsoft.operationalinsights/workspaces/yourLogAnalyticsWorkspaceName
+```
+
+#### Errors
+
+However, this Bicep fails and gets the following error, so I can't figure out how to deploy this setting via ARM/Bicep.
+All the docs  say to do this manually... I can't find how to automate this setting!
```
Comment (BernieWhite) [inline docs/en/rules/Azure.FrontDoor.Logs.md:115]: Try the suggestions above. A full sub-resource for key vault would look like:

```bicep
resource logs 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
  name: 'service'
  scope: vault
  properties: {
    workspaceId: workspaceId
    logs: [
      {
        category: 'AuditEvent'
        enabled: true
      }
    ]
  }
}
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@lluppesms Nice work on the examples! I've added a few suggestions below. Let me know if you have any questions.

File: docs/en/rules/Azure.FrontDoor.Logs.md [hunk 4]
```diff
@@ -25,6 +25,103 @@ Management operations for Front Door is captured automatically within Azure Acti
...
+```bicep
+'.../resourcegroups/myResourceGroupName/providers/Microsoft.Cdn/profiles/myFrontDoorName/providers/Microsoft.Insights/diagnosticSettings/service' is not supported"
+```
+
+---
+
 ## LINKS
 
 - [Monitoring metrics and logs in Azure Front Door Service](https://docs.microsoft.com/azure/frontdoor/front-door-diagnostics#diagnostic-logging)
+- [Create a Front Door Standard/Premium using Bicep](https://learn.microsoft.com/en-us/azure/frontdoor/create-front-door-bicep?tabs=CLI)
```
[the remark this answers is not in the document]
Comment (lluppesms) [reply docs/en/rules/Azure.FrontDoor.Logs.md:126]: Great suggestion!

File: docs/en/rules/Azure.FrontDoor.Logs.md (hunk 2 above)
Comment (lluppesms) [reply docs/en/rules/Azure.FrontDoor.Logs.md:84]: That looks good, but when I actually put that into a Bicep template it won't compile. "The property "scope" is not allowed on objects of type "microsoft.insights/diagnosticSettings". Permissible properties include "dependsOn"."

Comment (lluppesms) [reply docs/en/rules/Azure.FrontDoor.Logs.md:84]: I've tried many different ways, none of which worked... 

scope (none) => Scope "resourceGroup" is not valid for this resource type. Permitted scopes: "managementGroup"

scope: managementGroup(managementGroupIdentifier) => A resource's scope must match the scope of the Bicep file for it to be deployable. You must use modules to deploy resources to a different scope

scope: frontDoorResource  => Scope "resource" is not valid for this resource type. Permitted scopes: "managementGroup"

scope: 'managementGroup' => The property "scope" is not allowed on objects of type "microsoft.insights/diagnosticSettings". Permissible properties include "dependsOn"

Can't seem to find that secret sauce to get this to compile yet...

File: docs/en/rules/Azure.FrontDoor.Logs.md (hunk 3 above)
Comment (lluppesms) [reply docs/en/rules/Azure.FrontDoor.Logs.md:115]: I can get this to compile for the key vault, but the Front Door resource is scoped differently, so it throws an error with this syntax when I try to compile the bicep.

Comment (BernieWhite) [reply docs/en/rules/Azure.FrontDoor.Logs.md:115]: @lluppesms Maybe try to be explicit in the default scope of the module.

Have a look at this full deployment for front door:

```bicep
// Create or update a Front Door
targetScope = 'resourceGroup'

// ----------
// PARAMETERS
// ----------

@description('The name of the Front Door.')
param name string = deployment().name

@allowed([
  'Premium'
  'Standard'
])
@description('The SKU of the Front Door.')
param sku string = 'Standard'

@description('Configure TLS certificates for custom domains.')
param certificates array = []

@metadata({
  strongType: 'Microsoft.OperationalInsights/workspaces'
  example: '/subscriptions/<subscription_id>/resourceGroups/<resource_group>/providers/Microsoft.OperationalInsights/workspaces/<workspace_name>'
})
@description('The resource ID to a Log Analytics workspace.')
param workspaceId string = ''

@description('Configures the log categories to send to the Log Analytics workspace.')
param diagnosticLogs array = [
  'FrontDoorAccessLog'
]

@description('Tags to apply to the resource.')
param tags object = {}

// ---------
// VARIABLES
// ---------

var allTags = union(resourceGroup().tags, tags)
var skuName = '${sku}_AzureFrontDoor'

// ---------
// RESOURCES
// ---------

@description('Create or update a Front Door.')
resource profile 'Microsoft.Cdn/profiles@2021-06-01' = {
  name: name
  location: 'Global'
  sku: {
    #disable-next-line BCP036
    name: skuName
  }
  properties: {
    originResponseTimeoutSeconds: 60
  }
  tags: allTags
}

@description('Configure diagnostics logs to send to a workspace.')
resource logs 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (!empty(workspaceId) && !empty(diagnosticLogs)) {
  name: 'logs'
  scope: profile
  properties: {
    workspaceId: workspaceId
    logs: [for item in diagnosticLogs: {
      category: item
      enabled: true
    }]
  }
}

@description('Configure TLS certificates for custom domains.')
resource certificate 'Microsoft.Cdn/profiles/secrets@2021-06-01' = [for item in certificates: {
  parent: profile
  name: '${split(item, '/')[8]}-${last(split(item, '/'))}'
  properties: {
    parameters: {
      type: 'CustomerCertificate'
      secretSource: {
        id: item
      }
      useLatestVersion: true
    }
  }
}]

// -------
// OUTPUTS
// -------

@description('A unique identifier for the Front Door.')
output id string = profile.id

@description('The name of the Resource Group where the Front Door is deployed.')
output resourceGroupName string = resourceGroup().name

@description('The guid for the subscription where the Front Door is deployed.')
output subscriptionId string = subscription().subscriptionId

@description('A unique guid for the Front Door included in forwarded headers.')
output frontDoorId string = profile.properties.frontDoorId

```

**Comment by BernieWhite** [comment]
@lluppesms Looks like this is a duplicate of #1678. Should be close this PR?

**Comment by lluppesms** [comment]
Draft changes, replaced by #1678

**Pull request closed without merging**